tests/sles4sap/monitoring_services.pm

```diff
@@ -77,26 +79,45 @@ sub configure_hanadb_exporter {
         add_file_in_csync(value => "$hanadb_exporter_config");
 
         # Add the monitoring resource
-        assert_script_run(
-            'crm configure primitive ' . $hanadb_exp_rsc . " systemd:prometheus-$exporter_name@" . $args{rsc_id}
+        assert_script_run('crm configure primitive '
+              . $hanadb_exp_rsc
+              . " systemd:prometheus-$exporter_name@"
+              . $args{rsc_id}
               . ' op start interval=0 timeout=100'
               . ' op stop interval=0 timeout=100'
               . ' op monitor interval=10'
-              . ' meta target-role=Stopped'
-        );
+              . ' meta target-role=Stopped');
 
         assert_script_run "crm configure colocation col_exporter_$args{rsc_id} +inf: $hanadb_exp_rsc:Started $hanadb_msl:Master";
         assert_script_run "crm resource start $hanadb_exp_rsc";
         wait_until_resources_started;
-    } elsif (!get_var('HA_CLUSTER')) {
+    }
+    elsif (!get_var('HA_CLUSTER')) {
         $check_exporter = 'true';
         systemctl "enable --now prometheus-$exporter_name\@$args{rsc_id}";
     }
 
     # Check that the exporter is running
     if ($check_exporter eq 'true') {
         systemctl "status prometheus-$exporter_name\@$args{rsc_id}";
-        assert_script_run "curl -o $metrics_file http://localhost:$hanadb_exporter_port";
+        my $retry = 0;
+        my $count = 10;
+        while ($retry < $count) {
+            my $ret = script_run("curl -o $metrics_file http://localhost:$hanadb_exporter_port");
+            if ($ret) {
+                sleep 5;
+                record_info("Retry port $hanadb_exporter_port", script_output("lsof -i :$hanadb_exporter_port", timeout => 120, proceed_on_failure => 1));
+            }
+            else {
+                last;
+            }
```

Comment:
This could've been:
```
my $ret = script_run("curl -o $metrics_file http://localhost:$hanadb_exporter_port");
last if !$ret;
sleep 5;
record_info("Retry port $hanadb_exporter_port", script_output("lsof -i :$hanadb_exporter_port", timeout => 120, proceed_on_failure => 1));
```
And you avoid the extra indentation cause by the if-block.

Reply:
Thanks for the suggestion. But I have already merged this PR since I got 2 approvals.